### PR TITLE
start codifying db schema with migrations

### PR DESCRIPTION
**Pull request by andrew-nowak**
start a directory of DB migrations, so the schema is version controlled and can be iterated on and applied to different environments safely.

I've struggled and generally been dissatisfied with play evolutions in previous projects, as it's generally designed to work with runtime dependency injection rather than our preferred usage of compile-time/explicit injection, so I'm going back to flyway; somewhat annoying that it's a new dependency, but it can be run and applied independently of the main app, and is much easier to configure, which I find a positive, and generally doesn't try to lead you down the minefield of building "down" migrations

### CODE REVIEW

File: db/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# db
+
+This directory contains database configuration, including and especially the database migrations.
+
+Run the migrations with [flyway](https://documentation.red-gate.com/flyway/flyway-cli-and-api/welcome-to-flyway), which can be installed with Homebrew.
+
+This directory includes config for connecting to the local database, and is run as part of the start script.
+
+`flyway migrate`
```
Comment (bryophyta) [inline db/README.md:9]: I'm not quite clear about this -- is it saying that we should run `flyway migrate` directly? If so, under what circumstances?

File: db/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# db
+
+This directory contains database configuration, including and especially the database migrations.
+
+Run the migrations with [flyway](https://documentation.red-gate.com/flyway/flyway-cli-and-api/welcome-to-flyway), which can be installed with Homebrew.
+
+This directory includes config for connecting to the local database, and is run as part of the start script.
```
Comment (twrichards) [inline db/README.md:7]:
```suggestion
This directory includes config for connecting to the local database, and is run as part of the local dev start script.
```

**Review by bryophyta — APPROVED** [review]
one small comment about docs but I've run this locally (after deleting my previous local Docker container) and it all seems to work as expected 👍

File: db/migrations/V2__trigram_indices.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+CREATE INDEX headline_trgm_index ON fingerpost_wire_entry
+  USING GIN ((content->>'headline') gin_trgm_ops);
+
+CREATE INDEX subhead_trgm_index ON fingerpost_wire_entry
+  USING GIN ((content->>'subhead') gin_trgm_ops);
+
+CREATE INDEX keywords_trgm_index ON fingerpost_wire_entry
+  USING GIN ((content->>'keywords') gin_trgm_ops);
+
```
[the remark this answers is not in the document]
Comment (andrew-nowak) [reply db/migrations/V2__trigram_indices.sql:13]: We tried this on the CODE database, somewhat surprisingly (at least to me) this index and query operator continues to work even on the JSON array!

**Pull request merged**